inc/class_parser.php

```diff
@@ -1891,4 +1914,87 @@ function encode_url($url)
 
 		return $url;
 	}
+
+	/**
+	 * Determines whether the resulting HTML syntax is acceptable for output,
+	 * according to the parser's validation policy and HTML support.
+	 *
+	 * @param string $source The original MyCode.
+	 * @param string $output The output HTML code.
+	 * @return bool
+	 */
+	function output_allowed($source, $output)
+	{
+		if($this->output_validation_policy === 0 || !empty($this->options['allow_html']))
+		{
+			return true;
+		}
+		else
+		{
+			$output_valid = $this->validate_output($source, $output);
+
+			if($this->output_validation_policy === 1)
+			{
+				return true;
+			}
+			else
+			{
+				return $output_valid === true;
+			}
+		}
+	}
+
+	/**
+	 * Validate HTML syntax and pass errors to the error handler.
+	 *
+	 * @param string $source The original MyCode.
+	 * @param string $output The output HTML code.
+	 * @return bool
+	 */
+	function validate_output($source, $output)
+	{
+		global $mybb, $error_handler;
+
+		$ignored_error_codes = array(
+			'XML_ERR_INVALID_CHAR' => 9,
+			'XML_ERR_UNDECLARED_ENTITY' => 26,
+			'XML_ERR_ATTRIBUTE_WITHOUT_VALUE' => 41,
+			'XML_ERR_TAG_NAME_MISMATCH' => 76, // the parser may output tags closed in different levels and siblings
+		);
+
+		libxml_use_internal_errors(true);
```

Comment:
Do our system requirements already cover the requirement for `libxml`?

Reply:
It should be enabled by default, and may be satisfied by packages needed for SimpleXML